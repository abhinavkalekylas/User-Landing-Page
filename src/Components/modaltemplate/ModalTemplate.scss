.title {
    font-size: 2rem;
    margin-left: auto;
}

.modal-backdrop.show {
    opacity: 0.11;
}