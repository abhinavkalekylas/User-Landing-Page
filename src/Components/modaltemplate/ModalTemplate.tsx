import React from "react";
import { Modal, Button } from "react-bootstrap";
import "./ModalTemplate.css";

const ModalTemplate = ({
  title_name,
  userModal,
  closeModal,
  children,
}: {
  title_name: string;
  userModal: boolean;
  closeModal: () => void;
  children: any;
}) => {
  return (
    <Modal show={userModal} onHide={closeModal}>
      <Modal.Header closeButton>
        <Modal.Title className="title text-black">{title_name}</Modal.Title>
      </Modal.Header>
      <Modal.Body>{children}</Modal.Body>
      <Modal.Footer>
        <Button className="close_btn" variant="primary" onClick={closeModal}>
          Close
        </Button>
      </Modal.Footer>
    </Modal>
  );
};

export default ModalTemplate;
