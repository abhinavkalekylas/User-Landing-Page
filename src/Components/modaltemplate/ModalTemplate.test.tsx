import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ModalTemplate from "./ModalTemplate";

let wrapper: any;
beforeEach(() => {
  const props = {
    title_name: "Add User",
    userModal: true,
    closeModal: jest.fn(),
    children: React.Children,
  };
  wrapper = shallow(<ModalTemplate {...props} />);
});

// snapshot testing
it("renders correctly Modaldata component", () => {
  expect(toJson(wrapper)).toMatchSnapshot();
});

//test Close button
it("test close button", () => {
  wrapper.find(".close_btn").simulate("click");
  expect(
    wrapper.props().children[2].props.children.props.onClick.mock
      .invocationCallOrder[0]
  ).toEqual(1);
});
