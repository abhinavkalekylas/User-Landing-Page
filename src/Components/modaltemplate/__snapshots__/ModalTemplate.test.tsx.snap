// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Modaldata component 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "ModalDialog",
      "render": [Function],
    }
  }
  enforceFocus={true}
  keyboard={true}
  onHide={[MockFunction]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      className="title text-black"
    >
      Add User
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <Component />
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      className="close_btn"
      disabled={false}
      onClick={[MockFunction]}
      variant="primary"
    >
      Close
    </Button>
  </ModalFooter>
</Modal>
`;
